### PR TITLE
fix: avoid race conditions and concurrent updates on annotation update

### PR DESCRIPTION
**Pull request by PJEstrada**
To reduce cache errors occurrences:

1. Added the with_for_update() to instance list
2. Await the save method on frame changes instead of a sleep() time as sometimes the sleep can be less than the time the update takes, specially on slow connections.

### CODE REVIEW

File: frontend/src/components/video/video.vue [hunk 1]
```diff
@@ -693,11 +696,7 @@ export default Vue.extend( {
        *    but the channel between video and annotation core is not great
        */
 
-      this.$emit('request_save')
-
-      if (this.has_changed == true) {
-        await this.sleep(200) // give it a chance to save / fire off thing
-      }
+      await this.$props.parent_save()
```
Comment (anthony-sarkis) [inline frontend/src/components/video/video.vue:699]: Thanks goodness!

**Comment by PJEstrada** [comment]
Hey @anthony-sarkis ! I've re requested review, just so you can check the file lock code. I added the defaults for now which is skip_locked = False and nowait = False. That way the query just waits until the file is available with no errors. Let me know what you think about this, and if you feel the nowait should be change to throw an error we can def change it :)

**Comment by anthony-sarkis** [comment]
Thanks Pablo!  Besides item discussed in enterprise thread I think this
looks all good to me!

File: frontend/src/components/annotation/annotation_core.vue [hunk 2]
```diff
@@ -6268,6 +6269,9 @@ export default Vue.extend( {
         nodes: nodes,
         edges: edges,
         version: undefined,
+        root_id: undefined,
```
Comment (anthony-sarkis) [inline frontend/src/components/annotation/annotation_core.vue:6272]: Are there any other things we need to clear? What about creation_ref?

**Comment by PJEstrada** [comment]
Ok so this should be good now :) will wait for the tests and deploy as soon as possible

**Comment by anthony-sarkis** [comment]
Cool! :)

**Comment by anthony-sarkis** [comment]
Thank you!

**Pull request merged**